planner/core/find_best_task.go

```diff
@@ -572,18 +561,23 @@ func (ds *DataSource) getIndexMergeCandidate(path *util.AccessPath) *candidatePa
 
 // skylinePruning prunes access paths according to different factors. An access path can be pruned only if
 // there exists a path that is not worse than it at all factors and there is at least one better factor.
-func (ds *DataSource) skylinePruning(prop *property.PhysicalProperty) []*candidatePath {
+// The first return value indicates the candidate paths for CBO path selection.
+// The second return value indicates which paths are pruned if ds.ctx.GetSessionVars().StmtCtx.InVerboseExplain.
+func (ds *DataSource) skylinePruning(prop *property.PhysicalProperty) ([]*candidatePath, string) {
```

Comment:
How about just returning `prunedPaths []AccessPath` directly instead of a string?

Reply:
After discussing with @winoros, it seems better to output available candidates after pruning rather than pruned paths. In this way, it is easier to figure out which candidates participate in CBO. Hence we only need to return `[]*candidatePath` as before.